ports/stm32/uart.c

```diff
@@ -258,6 +293,7 @@ STATIC bool uart_init2(pyb_uart_obj_t *uart_obj) {
             irqn = USART6_IRQn;
             pins[0] = MICROPY_HW_UART6_TX;
             pins[1] = MICROPY_HW_UART6_RX;
+#if !defined(STM32F4)
```

Comment:
Why is this added?

Reply:
When looking at the configurations of all the permissible UARTS on the various `STM32`'s not all ports support hardware handshaking. So for the ports that don't have it, I excluded them in this way, but I imagine that when you leave out the `MICROPY_HW_UARTx_CTS`/`RTS` in the `mpconfigboard.h` it will not give a problem, however It is intended to be used as a guard to not generate code that cannot work with this MCU.